src/logger/MetricsContext.ts

```diff
@@ -111,6 +113,11 @@ export class MetricsContext {
    * @param dimensions
    */
   public putDimensions(incomingDimensionSet: Record<string, string>): void {
+    if (Object.keys(incomingDimensionSet).length > Constants.MAX_DIMENSION_SET_SIZE) {
+      throw new DimensionSetExceededError(
```

Comment:
Suggestion: Can we put this code behind a `validateDimensionSet` method? That way you can avoid duplication of this check and the exception.